### ISSUE TITLE
Switch to RFC 3986 compliant URL encoding for imgix

### ISSUE THREAD

**Issue by Mosnar** [aelvan/Imager-Craft#190]
The ImgixTransformer is using the non-RFC3986 compliant url encoding function. urlencode() is based on the original RFC written by Tim Burners Lee. This may have changed depending on PHP version used; however, testing on PHP 7.1.16 results in issues. This results in valid characters like the "@" symbol being transformed erroneously, resulting in 404s for S3 based Imgix configurations.

Employing a suggestion by a PHP commenter on the [urlencode documentation](http://php.net/manual/en/function.urlencode.php#97969) seems to be a viable solution:
`ImgixTransformer.php:382`
```php
    /**
     * URL encode the asset path properly
     *
     * @param string $path
     *
     * @return string
     */
    private function getUrlEncodedPath($path): string
    {
        $entities = array('%21', '%2A', '%27', '%28', '%29', '%3B', '%3A', '%40', '%26', '%3D', '%2B', '%24', '%2C', '%2F', '%3F', '%25', '%23', '%5B', '%5D');
        $replacements = array('!', '*', "'", "(", ")", ";", ":", "@", "&", "=", "+", "$", ",", "/", "?", "%", "#", "[", "]");
        $path = str_replace($entities, $replacements, urlencode($path));
        return $path;
    }
```

Based on my brief research, urlencode() seems to be a sticky subject, so open to a discussion. All I know is that my `@2x` imgix URLs are getting rejected as 404s.

Imgix Version: 2.1.0
PHP: 7.1.16
Craft: 3.0.22

**Comment by stursby** [comment]
@Mosnar @aelvan I think I was running into a similar issue. I'm using Imgix and the URLs that Imager returns have the params separated by `&amp;` which Imgix wasn't reading correctly. But if I change them to plain `&`, it all works.

Example

```twig
<meta content="{{ transformedImage.url|replace('&amp;', '&')|raw }}" name="twitter:image" />
```

@Mosnar I'm assuming that's what you meant with this Issue? I see the code in the lib where it's using `urlencode`...

![image](https://user-images.githubusercontent.com/527849/47666390-6fe70a80-db71-11e8-89fd-67d65cbe4182.png)

**Comment by stursby** [comment]
I just realized that Imager w/ Imgix is working fine as expected in my templates, but my issue from above was for meta images and how Craft was spitting out the URL. So really, I only need `|raw` for the meta tag image URLs.

**Comment by aelvan** [comment]
This was implemented as per @Mosnar's suggestion in v2.1.4.
